### PR TITLE
Removed DEBUG logging from PR builder profile

### PR DESCRIPTION
**Pull request by Donnerbart**
* added `log4j2-info.xml` without debug loggers to the project (for core and client module)
* made use of `log4j2-info.xml` in the `parallelTest` profile, which is used for the PR builder

The motivation behind this, is to give us more headroom on CloudBees regarding our disk quota. The PR build results are taking up almost 1 GB per build, which is mostly influenced by logging. In 2017 we increased the number of tests by ~25% (from 17.2k to 21.5k Quick + Slow tests), we also have an increase in the log size of ~18.5% in the PR builds (just Quick tests). This leads to a lot of "out of disk space" issues, since we easily can fill up all free disk space within a busy day. Actually the more we push, the faster we run into the issue. Our disk space is not scaling with our work output ;) CloudBees disk space is quite expensive, so striping down the log volume seems to be a considerable move.

Some additional justification, why I don't think it hurts us to lose the DEBUG info in the PR builder:
* The purpose of the PR builder is to give us fast feedback **if** a PR breaks something. Normally you analyse the **why** locally.
* You can always add a DEBUG logger to the `log4j2-info.xml` if needed.
* The `Test Result` pages on Jenkins never showed the DEBUG logs, just INFO or more severe. You always had to download the logs from S3, which I don't think a lot of developers do in the first place.

I checked the uncompressed logs from some PR builder runs, and it looks like we are down to ~425 from ~941 MB per build:
```
425M	new-lab-fast-pr-11657
427M	new-lab-fast-pr-11664
425M	new-lab-fast-pr-11665
425M	new-lab-fast-pr-11666
434M	new-lab-fast-pr-11667
422M	new-lab-fast-pr-11668
424M	new-lab-fast-pr-11669
```
I also checked that the configuration file is picked up, by adding a static prefix `foobar` to the logging pattern (build 11667):
```
foobar 09:42:14,966  INFO |aggregate_emptyFirstArray| - [system] aggregate_emptyFirstArray - [127.0.0.1]:5061 [dev] [3.10-SNAPSHOT] Hazelcast 3.10-SNAPSHOT (20171117 - 4b2e97f) starting at [127.0.0.1]:5061
```

### CODE REVIEW

**Comment by Donnerbart** [comment]
run-lab-run

**Comment by Donnerbart** [comment]
run-lab-run

**Comment by Donnerbart** [comment]
run-lab-run

**Comment by Donnerbart** [comment]
run-lab-run

File: hazelcast-client/src/test/resources/log4j2-info.xml [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+  -->
+
+<Configuration status="ERROR">
+    <Appenders>
+        <Console name="Console" target="SYSTEM_OUT">
+            <PatternLayout pattern="%d{ABSOLUTE} %5p |%X{test-name}| - [%c{1}] %t - %m%n"/>
+        </Console>
+    </Appenders>
+    <Loggers>
+        <Root level="INFO">
```
Comment (kwart) [inline hazelcast-client/src/test/resources/log4j2-info.xml]: Nitpicking :) Could you unify the letter-case for log levels in the PR?

Comment (Donnerbart) [reply hazelcast-client/src/test/resources/log4j2-info.xml]: Sure. I guess `UPPERCASE` is preferred?

Comment (kwart) [reply hazelcast-client/src/test/resources/log4j2-info.xml]: It depends.
Majority of log4j 2 configuration XML snippets in their documentation uses lowercase. For us, the old schoolers, uppercase is easier to read (as we don't need glasses for it). :)

**Pull request merged**